### PR TITLE
Vberenz/array

### PR DESCRIPTION
**Pull request by vincentberenz**
- implementation of array
- allows users to set size of memory segments

### CODE REVIEW

File: demos/demo_read_array.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    }
+  
+}
+
+
+int main()
+{
+  
+  // stop on ctrl+c
```
Comment (MaximilienNaveau) [inline demos/demo_read_array.cpp]: ignore for PR: in blmc_drivers I have put these in some header for all the demos as well as the global `RUNNING`, maybe we could do that here as well.

File: include/shared_memory/array.hpp [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+{
+  /**
+   * wipe the shared memory segment created by an instance
+   * of shared_memory::array, including mutexes, if any.
+   * If there are no memory segment of this id, there will be
+   * no effect. If shared_memory::array instances are pointing to the
+   * wiped out segment, their get and set functions may hang 
+   * indefinitely.
+   */
+  void clear_array(std::string segment_id);
```
Comment (MaximilienNaveau) [inline include/shared_memory/array.hpp]: so their is one array per segment?

File: include/shared_memory/shared_memory.hxx [hunk 3]
```diff
@@ -40,7 +40,7 @@ namespace shared_memory {
 
     if (registered){
```
Comment (MaximilienNaveau) [inline include/shared_memory/shared_memory.hxx]: Where is the Verbose?

File: tests/unit_tests.cpp [hunk 4]
```diff
@@ -648,5 +652,216 @@ TEST_F(Shared_memory_tests,serialization){
   ASSERT_EQ(in2.same(out2),true);
 
 }
+
+
+TEST_F(Shared_memory_tests,serialization2){
```
Comment (MaximilienNaveau) [inline tests/unit_tests.cpp]: Can you rename `Shared_memory_tests` to `SharedMemoryTests` to fit the cpp guidelines?

File: benchmarks/array.cpp [hunk 5]
```diff
@@ -0,0 +1,207 @@
+#include "shared_memory/array.hpp"
+#include "shared_memory/demos/item.hpp"
+#include <chrono>
+
+
+#define SEGMENT "benchmark_array"
+#define NB_ITERATIONS 250000
+
+
+class FrequencyMeasure
```
[the remark this answers is not in the document]
Comment (vincentberenz) [reply benchmarks/array.cpp]: I removed this file, it was not up to date with final API, sorry for this

File: include/shared_memory/array.hpp (hunk 2 above)
Comment (vincentberenz) [reply include/shared_memory/array.hpp]: yep

File: include/shared_memory/shared_memory.hxx (hunk 3 above)
Comment (vincentberenz) [reply include/shared_memory/shared_memory.hxx]: I hope it will come back with the merge with master

File: tests/unit_tests.cpp (hunk 4 above)
Comment (vincentberenz) [reply tests/unit_tests.cpp]: +1

**Pull request merged**